Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp

```diff
@@ -42,28 +41,29 @@ namespace Multiplayer
         // - LegacySystemInterfaceCreated is signaled, so that the logging system is ready. Automated testing listens for these logs.
         // - LegacyCommandLineProcessed is signaled, so that everything has initialized and finished their blocking loads, so that it
         // should be relatively safe to start receiving packets without as much fear of too much time passing between system ticks.
-        if (editorsv_isDedicated)
+#if AZ_TRAIT_SERVER && !AZ_TRAIT_CLIENT
```

Comment:
The Mutliplayer Sample Game Lift Gem already does something similar:
```
#define AZ_DEDICATED_SERVER_ONLY (AZ_TRAIT_SERVER && !AZ_TRAIT_CLIENT)
```

https://github.com/o3de/o3de-multiplayersample/blob/dd1c6e1b1bc5f3aca2d5b70f7b4c4687271463d8/MPSGameLift/Code/Source/MPSGameLiftModuleInterface.h#L18-L19

Reply:
I reverted this change back to the original because `editorsv_isDedicated` is currently necessary to toggle the editor being a dedicated server at runtime. I wanted to try to get rid of the CVAR, but without it, I'd have to set whether it is or is not one at compile time.